rcl/src/rcl/subscription.c

```diff
@@ -72,7 +72,8 @@ rcl_subscription_init(
   rcutils_string_map_t substitutions_map = rcutils_get_zero_initialized_string_map();
   rcutils_ret_t rcutils_ret = rcutils_string_map_init(&substitutions_map, 0, rcutils_allocator);
   if (rcutils_ret != RCUTILS_RET_OK) {
-    RCL_SET_ERROR_MSG(rcutils_get_error_string().str);
```

Comment:
I think the changes in this file should go in a separate PR since they resolve bugs and will need to be backported.

Reply:
That's fair point, though at the same time we need the additional tests to ensure those changes have the desired effect. What about splitting the PR in two commits that can be rebased instead of squashed? Once we've settled reviews, of course.

Reply:
> If you feel strongly that this should be done by squashing and splitting commits in this PR

No, I explicitly said rebasing, not squashing. Anyhow, if lack of tests in said PR is not a concern, I may as well go with it.

Reply:
See #739.

Reply:
> No, I explicitly said rebasing, not squashing

Sorry, I thought you were going to squash the 10 commits here into two separate commits then rebase on to master, one commit which would be cherry-pickable for backporting, and the other for the unit tests.